var/spack/repos/builtin/packages/py-pymatgen/package.py

```diff
@@ -15,21 +15,29 @@ class PyPymatgen(PythonPackage):
     homepage = "http://www.pymatgen.org/"
     pypi = "pymatgen/pymatgen-4.7.2.tar.gz"
 
+    version('2022.0.14', sha256='03d24ebafc21becab376c26de09437dd5cfb7bda7099046194e95b4c9fa35209')
+    version('2021.3.5',  sha256='de1f65dd3f4b1eb1626062865da29aa556c1796d93b9289135e0dc64a0adb286')
     version('4.7.2', sha256='e439b78cc3833a03963c3c3efe349d8a0e52a1550c8a05c56a89aa1b86657436')
     version('4.6.2', sha256='f34349090c6f604f7d402cb09cd486830b38523639d7160d7fd282d504036a0e')
 
     extends('python', ignore='bin/tabulate')
 
-    depends_on('py-setuptools@18.0:', type='build')
-
-    depends_on('py-numpy@1.9:',          type=('build', 'run'))
-    depends_on('py-six',                 type=('build', 'run'))
+    depends_on('python@3.5:',            when='@2021.3.5:')
+    depends_on('python@3.7:',            when='@2022.0.14:')
+    depends_on('py-cython',              type=('build'))
+    depends_on('py-setuptools@18.0:',    type=('build'))
+    depends_on('py-numpy@1.20.1:',       type=('build', 'run'))
     depends_on('py-requests',            type=('build', 'run'))
-    depends_on('py-pyyaml@3.11:',        type=('build', 'run'))
-    depends_on('py-monty@0.9.6:',        type=('build', 'run'))
-    depends_on('py-scipy@0.14:',         type=('build', 'run'))
-    depends_on('py-pydispatcher@2.0.5:', type=('build', 'run'))
+    depends_on('py-ruamel-yaml@0.15.6:', type=('build', 'run'))
+    depends_on('py-monty@3.0.2:',        type=('build', 'run'))
+    depends_on('py-scipy@1.5:',          type=('build', 'run'))
     depends_on('py-tabulate',            type=('build', 'run'))
-    depends_on('py-spglib@1.9.8.7:',     type=('build', 'run'))
+    depends_on('py-spglib@1.9.9.44:',    type=('build', 'run'))
+    depends_on('py-networkx@2.2:',       type=('build', 'run'))
     depends_on('py-matplotlib@1.5:',     type=('build', 'run'))
-    depends_on('py-palettable@2.1.1:',   type=('build', 'run'))
+    depends_on('py-palettable@3.1.1:',   type=('build', 'run'))
+    depends_on('py-sympy',               type=('build', 'run'))
+    depends_on('py-pandas',              type=('build', 'run'))
+    depends_on('py-plotly@4.5.0:',       type=('build', 'run'))
+    depends_on('py-uncertainties@3.1.4', type=('build', 'run'))
+    depends_on('py-netcdf4',             type=('test', 'run'))
```

Comment:
This dependency appears to be optional. I would either remove it or add a variant that defaults to False